src/pages/DogShow.js

```diff
@@ -1,9 +1,24 @@
 import React from 'react'
+import { useParams } from 'react-router-dom'
 
-const DogShow = () => {
+const DogShow = (props) => {
+  const { id } = useParams()
+ 
   return (
     <div>
-      <h1>DogShow</h1>
+      {props.dogs?.map((value, index) => {
+        if (+id === index){
```

Comment:
Keep in mind that id is referring to the dog id in params (not zero-indexed), whereas index is the location in the array (zero-indexed).   As it stands, your logic may provide some problems down the road.  If you are only showing one animal, I would strongly suggest using .find instead of map.